### PR TITLE
Added CustomArmor.

### PR DESCRIPTION
**Pull request by Wolftein**
Signed-off-by: Wolftein agustin.l.alvarez@hotmail.com

### CODE REVIEW

**Comment by ZNickq** [comment]
Please make sure it's finished, working, and tested before making the PR, also, when adding NMS classes for the first time, do it in a different commit, for diff visibility :P

File: src/minecraft/net/minecraft/src/SlotArmor.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+	/**
+	 * Check if the stack is a valid item for this slot. Always true beside for the armor slots.
+	 */
+	public boolean isItemValid(ItemStack par1ItemStack) {
+		//Spout start
+		if (par1ItemStack.itemID == 318) {
+			org.spoutcraft.spoutcraftapi.material.CustomItem item = MaterialData.getCustomItem(par1ItemStack.getItemDamage());
+			if( item != null )
+				return (item instanceof org.spoutcraft.spoutcraftapi.material.item.GenericCustomArmor);
```
Comment (ZNickq) [inline src/minecraft/net/minecraft/src/SlotArmor.java:39]: You can't put chestplates in the leggings section :P

File: src/minecraft/org/spoutcraft/client/packet/PacketGenericArmor.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+ */
+package org.spoutcraft.client.packet;
+
+import java.io.IOException;
+
+import org.spoutcraft.spoutcraftapi.io.SpoutInputStream;
+import org.spoutcraft.spoutcraftapi.io.SpoutOutputStream;
+import org.spoutcraft.spoutcraftapi.material.item.GenericCustomArmor;
+
+public class PacketGenericArmor implements SpoutPacket {
```
Comment (ZNickq) [inline src/minecraft/org/spoutcraft/client/packet/PacketGenericArmor.java:28]: This whole class doesn't do anything, is it a stub?

File: src/minecraft/net/minecraft/src/SlotArmor.java (hunk 1 above)
Comment (Wolftein) [inline src/minecraft/net/minecraft/src/SlotArmor.java:39]: Fixed!

File: src/minecraft/org/spoutcraft/client/packet/PacketGenericArmor.java (hunk 2 above)
Comment (Wolftein) [inline src/minecraft/org/spoutcraft/client/packet/PacketGenericArmor.java:28]: Bad commit, dunno why its not complete..

**Pull request closed without merging**